tests/lib/kubernetes/caasp.py

```diff
@@ -71,7 +72,21 @@ def join(self, node: NodeBase):
             capture=True, check=True, chdir=self._clusterpath
         )
 
+    def join(self, nodes: List[NodeBase]):
+        super().join(nodes)
+
+        self.hardware.ansible_run_playbook('playbook_caasp.yaml', nodes)
```

Comment:
This duplicates the step in `bootstrap`. I think it can be removed from bootstrap.

Reply:
it does not. `bootstrap()` is there for setting up the initial node. `join()` is there to add new nodes later.
So we need it in both functions. But in `bootstrap()`, we could limit the run to just the first master. But then the whole thing will be done later.
I wonder how we can optimize it....

Reply:
Does bootstrap need to do the worker nodes? We could just prepare the master there and then do the workers here.

Granted we lose Ansible being able to do the playbook in parallel.

Another option is to keep track of which nodes have been "prepared".

Or just let it run regardless as it should be idempotent.

Reply:
I think we should do in `bootstrap()` only the first master node. and then do the rest in `join()`. I don't like to track what was already prepared...

Reply:
Agreed. Did you want to update this here?

Reply:
nope. in another PR